kubernetes-informer/src/main/java/io/micronaut/kubernetes/client/informer/DefaultSharedIndexInformerFactory.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.informer;
+
+import com.google.gson.reflect.TypeToken;
+import io.kubernetes.client.apimachinery.GroupVersionKind;
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.informer.ListerWatcher;
+import io.kubernetes.client.informer.SharedIndexInformer;
+import io.kubernetes.client.informer.SharedInformerFactory;
+import io.kubernetes.client.informer.impl.DefaultSharedIndexInformer;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.util.CallGeneratorParams;
+import io.kubernetes.client.util.Namespaces;
+import io.kubernetes.client.util.Watchable;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.options.ListOptions;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.kubernetes.client.ModelMapper;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Default implementation of the {@link SharedIndexInformerFactory}. The created {@link SharedIndexInformer} is shared
+ * in scope of the namespace.
+ *
+ * @author Pavol Gressa
+ * @since 3.1
+ */
+@Requires(property = DefaultSharedIndexInformerFactory.INFORMER_ENABLED, notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
+@Singleton
+public class DefaultSharedIndexInformerFactory extends SharedInformerFactory implements SharedIndexInformerFactory {
+    public static final String INFORMER_ENABLED = "kubernetes.client.informer.enabled";
+    public static final Duration INFORMER_START_WAIT_TIME = Duration.ofMinutes(1);
+    public static final Duration INFORMER_START_STEP_TIME = Duration.ofMillis(500);
```

Comment:
Do these need to be configurable?

Reply:
Initially I thought it's not needed since the one minute limit should be enough, but for extreme cases when one minute timeout won't be enough maybe it will be handy to have the option to configure.